### PR TITLE
Lint the less files of the simple theme

### PR DESCRIPTION
**Pull request by return42**
## What does this PR do?

Lint the less files of the simple theme

About details read along the commit messages.

This PR is limited to the simple theme (I hope we can drop the oscar theme, see #162)

### CODE REVIEW

File: manage [hunk 1]
```diff
@@ -488,8 +488,12 @@ themes.oscar() {
 }
 
 themes.simple() {
-    build_msg GRUNT "theme: simple"
-    npm --prefix searx/static/themes/simple run build
+    (   set -e
+        build_msg LINT "theme: simple"
+        npm --prefix searx/static/themes/simple run stylelint
+        build_msg GRUNT "theme: simple"
+        npm --prefix searx/static/themes/simple run build
```
Comment (dalf) [inline manage]: `grunt build` already calls `jshint`: 
https://github.com/searxng/searxng/blob/056550d523ddf4e06821957e13623fc322cf4675/searx/static/themes/simple/gruntfile.js#L197

so I would add the call `stylelint` from `gruntfile.js` using [grunt-stylelint](https://www.npmjs.com/package/grunt-stylelint):
```js
grunt.registerTask('default', ['jshint', 'stylelint', 'copy', 'concat', 'uglify', 'less:development', 'less:production']); 
```

File: searx/static/themes/simple/package.json [hunk 2]
```diff
@@ -25,6 +27,7 @@
     "build": "grunt",
     "watch": "grunt watch",
     "webfont": "grunt webfont",
-    "clean": "rm -Rf node_modules package-lock.json ion.less"
+    "clean": "rm -Rf node_modules package-lock.json ion.less",
+    "stylelint": "stylelint -f unix 'src/less/**/*.less'"
```
Comment (dalf) [inline searx/static/themes/simple/package.json]: Based on the comment on `manage` :
```js
  "stylelint": "grunt stylelint",
  "stylelint-fix": "grunt --fix stylelint"
```


https://www.npmjs.com/package/grunt-stylelint#fix

File: searx/static/themes/simple/src/less/style.less [hunk 3]
```diff
@@ -704,7 +704,7 @@ article.result-images[data-vim-selected]::before {
 
   #linkto_preferences {
     display: none;
-    postion: fixed !important;
+    postion: fixed !important; // stylelint-disable property-no-unknown
```
Comment (dalf) [inline searx/static/themes/simple/src/less/style.less]: There is a typo `postion` --> `position` !

**Review by dalf — COMMENTED** [review]
Expect the 3 comments LGTM.

I've compiled the theme and look at the result: everything is okay.

File: manage (hunk 1 above)
Comment (return42) [reply manage]: I have seen that there is a grunt-stylelint, but it uses an old ["stylelint": "13.8.0"](https://github.com/wikimedia/grunt-stylelint/commit/5c07ec3767145aeafae043db8d5d3f95f055aec3) while I decided to use the latest versions of stylelint & stylelint-config-standard in our [package.json](https://github.com/searxng/searxng/pull/168/commits/e22dbdda53445008c6b089365c3ba152413a76bc#diff-0f087f6a54013c2b6e21074952c187930cd193591ec780748f18423271435f90).

Do you see a disadvantage in this? ..

Foregive me if I am naive and not used to work with grunt: My impression is that the grunt packages are partly outdated and not updated regularly, so I asking myself why not directly use the npm packages (npm run configuration).  What I mean: we have these shell scripts to build our product, do we realy need additionally grunt packages which wrap old libraries? .. is my thought wrong, what do you think?

File: searx/static/themes/simple/src/less/style.less (hunk 3 above)
Comment (return42) [reply searx/static/themes/simple/src/less/style.less]: LOL :-) .. here you can see that I am a dyslexic .. I rebased and dropped this commit 50495757, the fix is now in 5b7e7aa.

File: manage (hunk 1 above)
Comment (dalf) [reply manage]: The dependency is ["stylelint": "^13.8.0"](https://github.com/wikimedia/grunt-stylelint/blob/9e56310596ec39c587f5ef20361a73dd0be9fb40/package.json#L26)

See https://docs.npmjs.com/about-semantic-versioning#using-semantic-versioning-to-specify-update-types-your-package-can-accept

`npm install grunt-stylelint stylelint --save-dev` (from https://www.npmjs.com/package/grunt-stylelint), installs the last version:
```js
{
    "grunt-stylelint": "^0.16.0",
    "stylelint": "^13.13.1",
    "stylelint-config-standard": "^22.0.0"
}
```

Comment (return42) [reply manage]: > "stylelint": "^13.8.0" .. installs the last version

Argh, you are right .. I have overseen the `^` and thought they are using old versions .. sorry, I am unexercised with npm .. forget my comment, I will patch to use grunt.

File: searx/static/themes/simple/src/less/style.less (hunk 3 above)
Comment (dalf) [reply searx/static/themes/simple/src/less/style.less]: Actually I compiled the theme, start the devtools, saw the property was not recognized, and then I realized the problem was not the value but the property name...

File: manage (hunk 1 above)
Comment (return42) [reply manage:495]: Sorry for another dump question, just for my knowledge of npm;

> npm install grunt-stylelint stylelint --save-dev (from https://www.npmjs.com/package/grunt-stylelint), installs the last version:

What version of stylelint would be used, when I install only::

    npm install grunt-stylelint

When I look at the `package-lock.json` file I can't find an entry for stylelint / I also looked at the node_modules folder, there is no stylelint package.  This is what me wonders, I thought  `package-lock.json` describes the full dependencies.

Comment (return42) [reply manage]: Now I added a stylelint task::

```js
    stylelint: {
      options: {
        formatter: 'unix',
      },
      src: [
        'src/less/**/*.less',
      ]
    },
```

File: searx/static/themes/simple/package.json (hunk 2 above)
Comment (return42) [reply searx/static/themes/simple/package.json]: > `"stylelint-fix": "grunt --fix stylelint"`

It seems `"stylelint-fix": "grunt stylelint --fix"`did also work .. I haven't found a documentation which describes what arguments from the command line are passed through .. what I asking myself is what about common arguments like `--help` or `-v`, these are grunt options and stylint has also the same options.  I looked at:
 
- https://gruntjs.com/using-the-cli and
- https://gruntjs.com/api/grunt.option

**Comment by return42** [comment]
@dalf: I rebased, to get the line back ..

-  dropped this commit 5049575, the fix is now in https://github.com/searxng/searxng/commit/0b7d03c6#diff-88d28efcc4bd7593a1a47d69dfcb5f50c147b93fceb99c58d7ee3b731e01a1f9R707
- a3f58a3 is new since your last review
- dfc5c3bc has been modified to fit your request about [grunt](https://github.com/searxng/searxng/pull/168#discussion_r656784723)

The following patches has not been changed (except the SHA commit caused by the rebase)

- [fix] simple theme: use stylint to fix common lint errors
- [stylelint] simple theme: fix some errors reported by stylelint
- [stylelint] disable role 'no-descending-specificity'

> I've compiled the theme and look at the result: everything is okay.

I did not committed the "Static build" .. I will test this with your  #174 :-)

**Review by dalf — APPROVED** [review]

**Pull request merged**